packages/oui-radio-toggle/src/index.spec.js

```diff
@@ -0,0 +1,133 @@
+describe("ouiRadioToggle", () => {
+    let TestUtils;
+    let $timeout;
+    let $rootScope;
+
+    beforeEach(angular.mock.module("oui.radio"));
+    beforeEach(angular.mock.module("oui.radio-group"));
+    beforeEach(angular.mock.module("oui.radio-toggle"));
+    beforeEach(angular.mock.module("oui.test-utils"));
+
+    beforeEach(inject((_TestUtils_, _$timeout_, _$rootScope_) => {
+        TestUtils = _TestUtils_;
+        $timeout = _$timeout_;
+        $rootScope = _$rootScope_;
+    }));
+
+    const getRadioInputElement = (element) => angular.element(element[0].querySelector("input[type=radio]"));
+    const getRadioInputElementWithValue = (element, value) => angular.element(element[0].querySelector(`input[type=radio][value=${value}]`));
+    const getRadioGroupElement = (element) => angular.element(element[0]);
+    const clickRadio = (radioToCheck) => {
+        radioToCheck.prop("checked", true);
+        radioToCheck.triggerHandler("click");
+    };
+
+    describe("Component", () => {
+
+        describe("attributes", () => {
+
+            it("should assign defined name to child radios when name attribute is defined", () => {
+                const name = "foo";
+
+                const element = TestUtils.compileTemplate(`
+                    <oui-radio-toggle name=${name}>
+                        <oui-radio></oui-radio>
+                    </oui-radio-toggle>
+            `);
+
+                expect(getRadioInputElement(element).prop("name")).toBe(name);
+            });
+
+            fit("should assfsdfsign a generated name to child radios when name attribute is undefined ", () => {
+                console.log("begin0");
+                const element = TestUtils.compileTemplate(`
+                    <oui-radio-toggle>
+                        <oui-radio></oui-radio>
+                    </oui-radio-toggle>
+                `);
+
+                expect(getRadioInputElement(element).prop("name")).toMatch(/^oui-radio-toggle-\d+$/);
+                console.log("end");
```

Comment:
`console.log`

Reply:
This pull-request is only to propose a new pattern, it's not done.
No comment on the pattern?

Reply:
the line above it is fine for me, nice way to test the expected name for the generated one. But remove the `console.log` like @neolitec said ;)